test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -68,11 +70,32 @@
       app {:jruby-puppet
            {:max-active-instances num-jrubies}
            :versioned-code
-           {:code-id-command (script-path "warn_echo_and_error")}}
+           {:code-id-command (script-path "warn_echo_and_error")
+            :code-content-command (script-path "echo")}}
       (let [catalog (testutils/get-catalog)]
         (is (nil? (get catalog "code_id")))
         (is (logged? #"Non-zero exit code returned while running" :error))
-        (is (logged? #"Executed an external process which logged to STDERR: production" :warn)))))))
+        (is (logged? #"Executed an external process which logged to STDERR: production" :warn))))))
+  (testing "code id is not added and 500 is returned if code-content-command is not set"
+    (logging/with-test-logging
+     (bootstrap/with-puppetserver-running
+      app {:jruby-puppet
+           {:max-active-instances num-jrubies}
+           :versioned-code
+           {:code-id-command (script-path "echo")}}
+      (let [response (testutils/http-get "puppet/v3/catalog/localhost?environment=production")]
+        (is (= 500 (:status response)))
+        (is (logged? #"Error 500 on SERVER"))))))
```

Comment:
> The discussion in the meeting the other day mentioned that for any catalog compilation that is not using Direct Puppet, the catalog compilation should still succeed.

... even if `code-id-command` is present but `code-content-command` is not?  I don't remember that being stated, although it sounds totally reasonable, especially in a case like when `static_catalogs = false`.  However, if that's true, we definitely need to rethink our approach here.

> IIRC, the determination of whether we're using DP or not will be based on a combination of:
1. did the agent send up a query param indicating that they support static catalogs?
2. is the per-environment setting for static catalogs enabled for this environment?
> The first question should be easy to answer. 

Yes, it would be easy to answer, but AFAIK that's not in the plans for this iteration.

I agree that we should discuss this again.